Shop.BLL/Services/AuthService.cs

```diff
@@ -0,0 +1,69 @@
+using System.Text;
+using AutoMapper;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using Shop.BLL.Common.DataTransferObjects.Users;
+using Shop.BLL.Exceptions.InternalExceptions;
+using Shop.BLL.Interfaces;
+using Shop.DAL.Models;
+
+namespace Shop.BLL.Services
+{
+    public class AuthService : IAuthService
+    {
+        private readonly IMapper _mapper;
+        private readonly UserManager<User> _userManager;
+        private readonly ILogger<AuthService> _logger;
+
+        public AuthService(IMapper mapper, UserManager<User> userManager,
+            ILogger<AuthService> logger)
+        {
+            _mapper = mapper;
+            _userManager = userManager;
+            _logger = logger;
+        }
+        public async Task<bool> AuthorizeUserAsync(UserRequestAuthorizationDto userRequestAuthorizationDto)
+        {
+            var user = await _userManager
+                .FindByEmailAsync(userRequestAuthorizationDto.Email);
+
+            var res = user != null && await _userManager.CheckPasswordAsync(user,
+                userRequestAuthorizationDto.Password);
+
+            if (!res)
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
in this method, you need to create a token and return it

Reply:
It was discussed at daily. Token issuing perfomed by OpenIdDict using SignIn method with OpenIddictServerAspNetCoreDefaults.AuthenticationScheme